wire/msgtx.go

```diff
@@ -753,15 +772,33 @@ func writeTxIn(w io.Writer, pver uint32, version int32, ti *TxIn) error {
 
 // readTxOut reads the next sequence of bytes from r as a transaction output
 // (TxOut).
-func readTxOut(r io.Reader, pver uint32, version int32, to *TxOut) error {
+func readTxOut(r io.Reader, pver uint32, version int32, to *TxOut) (int, error) {
 	err := readElement(r, &to.Value)
 	if err != nil {
-		return err
+		return 0, err
 	}
+	var ScriptAndPossibleTokenData []byte
+	ScriptAndPossibleTokenData, err = readScript(r, pver, maxMessagePayload(),
+		"transaction output public key script and possible token data")
 
-	to.PkScript, err = readScript(r, pver, maxMessagePayload(),
-		"transaction output public key script")
-	return err
+	scriptAndTokendataSize := len(ScriptAndPossibleTokenData)
+
+	if scriptAndTokendataSize > 0 && ScriptAndPossibleTokenData[0] != 0xef { // Todo maybe change this
+		to.PkScript = ScriptAndPossibleTokenData
+		return scriptAndTokendataSize, err
+	}
+	var tokenDataSeparationErr error
+	to.PkScript, tokenDataSeparationErr = to.TokenData.SeparateTokenDataFromPKScriptIfExists(ScriptAndPossibleTokenData, pver)
+
+	if tokenDataSeparationErr != nil { // after activation this must fail! TODO TODO
```

Comment:
Something left to be done here?

Reply:
I think that comment is left from an earlier iteration. Can't think why it should exists.